lib/puppet/provider/ios_stp_global/cisco_ios.rb

```diff
@@ -207,4 +207,25 @@ def delete(context, _name, is)
       context.transport.run_command_conf_t_mode(command)
     end
   end
+
+  def canonicalize(context, resources)
+    resources.each do |resource|
+      current_instance = (get(context) || []).find { |key| key[:name] == resource[:name] }
+      # we will need to assume that
+      # ios_stp_global { 'default':
+      #   extend_system_id => true,
+      #   loopguard => false,
+      #   uplinkfast => false,
+      # }
+      # is considered the defaults and spt is disbaled
+      # if the current instance is these values
+      # then the enable value is no longer required
+      # from the resource
+      next unless current_instance
+      if current_instance[:extend_system_id] && !current_instance[:loopguard] && !current_instance[:uplinkfast]
+        resource.delete(:enable) unless resource[:enable].nil?
```

Comment:
As discussed, please change this to `get` returning an `enable` that actually makes sense given the rest of the implementation.